### PR TITLE
[ISSUE-257][FEATURE] Reserve slots support customized retry times

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### Related issues.
#257 

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture @FMX

File: client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala [hunk 1]
```diff
@@ -931,76 +931,81 @@ class LifecycleManager(appId: String, val conf: RssConf) extends RpcEndpoint wit
   }
 
   /**
-   * reserve buffer with retry, retry on another node will cause slots to be inconsistent
+   * Reserve buffers with retry, retry on another node will cause slots to be inconsistent.
```
Comment (AngersZhuuuu) [inline client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala:934]: `retry on another node will cause slots to be inconsistent` Want to know what it means.

**Comment by waitinfuture** [comment]
LGTM, but I suggest to conduct a e2e test since it's critical.

**Review by FMX — APPROVED** [review]
LGTM.

**Pull request merged**